File: models/Match.js
const { Model, DataTypes } = require('sequelize');
const sequelize = require('../config/database');

class Match extends Model {}

Match.init({
    id: {
        type: DataTypes.STRING,
        primaryKey: true
    },
    sport_type: {
        type: DataTypes.STRING,
        allowNull: false
    },
    home_team: {
        type: DataTypes.STRING,
        allowNull: false
    },
    away_team: {
        type: DataTypes.STRING,
        allowNull: false
    },
    home_team_color: {
        type: DataTypes.STRING,
        defaultValue: '#1e40af'
    },
    away_team_color: {
        type: DataTypes.STRING,
        defaultValue: '#1e40af'
    },
    home_team_header: {
        type: DataTypes.STRING,
        defaultValue: 'HOME'
    },
    away_team_header: {
        type: DataTypes.STRING,
        defaultValue: 'AWAY'
    },
    home_score: {
        type: DataTypes.INTEGER,
        defaultValue: 0
    },
    away_score: {
        type: DataTypes.INTEGER,
        defaultValue: 0
    },
    status: {
        type: DataTypes.STRING,
        defaultValue: 'pending'
    },
    match_data: {
        type: DataTypes.JSONB,  // PostgreSQL JSONB 타입으로 변경
        defaultValue: {}
    },
    created_by: {
        type: DataTypes.INTEGER,
        allowNull: true,
        references: {
            model: 'users',
            key: 'id'
        }
    }
}, {
    sequelize,
    modelName: 'Match',
    timestamps: true,
    createdAt: 'created_at',
    updatedAt: 'updated_at',
    tableName: 'Matches'
});

module.exports = Match; 

File: models/Match.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;
using Newtonsoft.Json;

namespace SportsCoderWinForm.Models
{
    [Table("Matches")]
    public class Match
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string SportType { get; set; } = string.Empty;

        [Required]
        [MaxLength(100)]
        public string HomeTeam { get; set; } = string.Empty;

        [Required]
        [MaxLength(100)]
        public string AwayTeam { get; set; } = string.Empty;

        public int HomeScore { get; set; } = 0;

        public int AwayScore { get; set; } = 0;

        [MaxLength(20)]
        public string Status { get; set; } = "pending";

        [Column(TypeName = "TEXT")]
        public string MatchDataJson { get; set; } = "{}";

        [NotMapped]
        public Dictionary<string, object> MatchData
        {
            get
            {
                try
                {
                    return JsonConvert.DeserializeObject<Dictionary<string, object>>(MatchDataJson) ?? new Dictionary<string, object>();
                }
                catch
                {
                    return new Dictionary<string, object>();
                }
            }
            set
            {
                MatchDataJson = JsonConvert.SerializeObject(value);
            }
        }

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        public DateTime UpdatedAt { get; set; } = DateTime.Now;
    }
} 